### PR TITLE
remove airbyte-commons-docker

### PR DESCRIPTION
**Pull request by colesnodgrass**
## What
- part of the repo split work; cleaning up code and dependencies so they won't need to be duplicated
- remove shared library
    - `getTaggedImage` was simply inserting a `:` between two strings, replaced with `+ ":" +`
    -  `buildImage` was used in one test class, copied the `DockerUtils` class to that test class
- Ideally a method `getImageTag` would be added to the `StandardSourceDefinition` and `StandardDestintationDefinition` classes that would return image-tag name, as these classes already know about the repository and the tag.  These classes are auto generated and I didn't see a way (in my quick search) to add a derived method attribute like this, as this wouldn't need to be part of the data of the object.

## How
- replace calls to shared library with non shared code
- had to add the BouncyCastle dependencies to the `airbyte-workers` test-implementation dependencies
    - tests started failing with `JcaPEMKeyConverter is provided by BouncyCastle, an optional dependency. To use support for EC Keys you must explicitly add this dependency to classpath` errors when the `airbyte-commons-docker` dependency was removed.  `airbyte-commons-docker` didn't depend on BouncyCastle, so I'm guessing this was being brought in via a transitive dependency that `airbyte-commons-docker` brought with it.

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -1159,7 +1158,7 @@ jobs:
   helm-acceptance-test:
     name: "Platform: Acceptance Tests (Helm)"
     # In case of self-hosted EC2 errors, removed the `needs` line and switch back to running on ubuntu-latest.
-    needs: [start-helm-acceptance-test-runner] # required to start the main job when the runner is ready
+    needs: [ start-helm-acceptance-test-runner ] # required to start the main job when the runner is ready
```
Comment (benmoriceau) [inline .github/workflows/gradle.yml:1161]: Is this change expected? Shouldn't it be
```
needs:
     - start-helm-acceptance-test-runner
```

**Review by benmoriceau — APPROVED** [review]

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (colesnodgrass) [reply .github/workflows/gradle.yml:1157]: This was the formatter adding some spaces between the square-braces.  From the yaml parsing POV, there isn't a difference between
```yml
needs: [ value ]
```
and 
```yml
needs:
  - value
```

File: airbyte-commons-server/src/test/java/io/airbyte/commons/server/handlers/SourceDefinitionsHandlerTest.java [hunk 2]
```diff
@@ -335,7 +334,7 @@ void testCreateSourceDefinitionWithInvalidProtocol() throws URISyntaxException,
     final String invalidProtocol = "131.1.2";
     final StandardSourceDefinition sourceDefinition = generateSourceDefinition();
     sourceDefinition.getSpec().setProtocolVersion(invalidProtocol);
-    final String imageName = DockerUtils.getTaggedImageName(sourceDefinition.getDockerRepository(), sourceDefinition.getDockerImageTag());
+    final String imageName = sourceDefinition.getDockerRepository() + ":" + sourceDefinition.getDockerImageTag();
```
[the remark this answers is not in the document]
Comment (colesnodgrass) [reply airbyte-commons-server/src/test/java/io/airbyte/commons/server/handlers/SourceDefinitionsHandlerTest.java:335]: Now that you mention this, I'm a little surprised it didn't complain, but 
image

[the remark this answers is not in the document]
Comment (benmoriceau) [reply airbyte-commons-server/src/test/java/io/airbyte/commons/server/handlers/SourceDefinitionsHandlerTest.java:335]: Yeah, I don't mind it too much for case like that.

**Pull request merged**